haystack/nodes/prompt/prompt_node.py

```diff
@@ -803,19 +811,48 @@ def is_supported_template(self, prompt_template: Union[str, PromptTemplate]) ->
         template_name = prompt_template if isinstance(prompt_template, str) else prompt_template.name
         return template_name in self.prompt_templates
 
-    def get_prompt_template(self, prompt_template: Union[str, PromptTemplate, None]) -> PromptTemplate:
+    def get_prompt_template(self, prompt_template: Union[str, PromptTemplate, None] = None) -> Optional[PromptTemplate]:
         """
-        Returns a prompt template by name.
-        :param prompt_template_name: The name of the prompt template to be returned.
-        :return: The prompt template object.
+        Resolves a prompt template.
+
+        :param prompt_template: The prompt template to be resolved. You can choose between the following types:
+            - None: returns the default prompt template
+            - PromptTemplate: returns the given prompt template object
+            - str: parses the string depending on its content:
+                - prompt template name: returns the prompt template registered with the given name
+                - prompt template yaml: returns a prompt template specified by the given yaml
+                - prompt text: returns a copy of the default prompt template with the given prompt text
+
+            :return: The prompt template object.
         """
+        prompt_template = prompt_template or self.default_prompt_template
+        if prompt_template is None:
+            return None
+
         if isinstance(prompt_template, PromptTemplate):
             return prompt_template
 
-        if not isinstance(prompt_template, str) or prompt_template not in self.prompt_templates:
-            raise ValueError(f"Prompt template {prompt_template} not supported")
+        if isinstance(prompt_template, str) and prompt_template in self.prompt_templates:
+            return self.prompt_templates[prompt_template]
 
-        return self.prompt_templates[prompt_template]
+        # if it's not a string or looks like a prompt template name
+        if not isinstance(prompt_template, str) or re.fullmatch(r"[-a-zA-Z0-9_]+", prompt_template):
+            raise ValueError(
+                f"{prompt_template} not supported, please select one of: {self.get_prompt_template_names()} or pass a PromptTemplate instance for prompting."
+            )
+
+        if "prompt_text:" in prompt_template:
+            prompt_template_parsed = yaml.safe_load(prompt_template)
+            if isinstance(prompt_template_parsed, dict):
+                return PromptTemplate(**prompt_template_parsed)
+
+        # it's a prompt_text
+        prompt_text = prompt_template
+        output_shapers = []
+        default_prompt_template = self.get_prompt_template()
+        if default_prompt_template:
+            output_shapers = default_prompt_template.output_shapers
+        return PromptTemplate(name="custom-at-query-time", prompt_text=prompt_text, output_shapers=output_shapers)
```

Comment:
This is the new resolution feature. If it's a string, but not a registered name then
- first check if it's a YAML definition (full definition of prompt template including output_shapers) and if so parse it
- else (we know it's not a YAML definition and not a registered name or lookalike, but a string) assume it's a prompt_text, so copy the default prompt template (i.e. output_shapers) and set the prompt_text

That way we enable setting `prompt_texts` and serialized PromptTemplates (prompt template definitions) at query time.